protocol/proto/types.go

```diff
@@ -60,9 +64,6 @@ type EncodeDecoder interface {
 
 	// Close closes the EncodeDecoder.
 	Close()
-
-	// Reset resets the EncodeDecoder.
-	Reset(rw io.ReadWriter)
```

Comment:
I remember the reason why you have this on the `EncoderDecoder` interface is that you need to reset the reader for both the encoder and the decoder at the same time, is that no longer needed?

Reply:
I used to have that so that the user can reset both the writer and reader in one call, but now the encoder longer references a writer, so that is no longer needed.